### PR TITLE
"CONTAINS" support for BKD-backed geo_shape and shape fields

### LINKED ISSUE #41204: CONTAINS query support for BKD-based geoshapes
`contains` relationship for the new BKD-based geoshapes is dependent on https://issues.apache.org/jira/browse/LUCENE-8620 merging into Lucene.

After that is merged, we need to expose this option in Elasticsearch and make it work with existing queries.  The `contains` relationship also needs to fail for indexes created in previous versions as the index does not contain enough information.

### PR DESCRIPTION
**Pull request by iverase**
Lucene 8.4 added support for "CONTAINS", therefore in this commit those
changes are integrated in Elasticsearch. This commit contains as well a
bug fix when querying with a geometry collection with "DISJOINT" relation.

Closes #41204

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/query/VectorGeoShapeQueryProcessor.java [hunk 1]
```diff
@@ -144,7 +149,13 @@ public Query visit(MultiPolygon multiPolygon) {
         @Override
         public Query visit(Point point) {
             validateIsGeoShapeFieldType();
-            return LatLonShape.newBoxQuery(fieldName, relation.getLuceneRelation(),
+            ShapeField.QueryRelation lucenRelation = relation.getLuceneRelation();
```
Comment (jpountz) [inline server/src/main/java/org/elasticsearch/index/query/VectorGeoShapeQueryProcessor.java]: missing `e`: `luceneRelation`

**Review by jpountz — COMMENTED** [review]
Do we need to disallow this relation on pre-7.6 indices?

File: server/src/main/java/org/elasticsearch/index/query/VectorGeoShapeQueryProcessor.java (hunk 1 above)
Comment (iverase) [reply server/src/main/java/org/elasticsearch/index/query/VectorGeoShapeQueryProcessor.java]: ups

**Comment by iverase** [comment]
I think we can perform the query as well in 7.5 indexes but not before.

**Comment by iverase** [comment]
I checked and yes, we started labelling the edges of a polygon in Lucene 8.3 so indexes created in Elasticsearch 7.5.x are fine.

I added the check but I need to work out how to test this feature. We can improve the error message as well.

**Review by jpountz — APPROVED** [review]

**Pull request merged**